### PR TITLE
refactor: change debounce function using rooks

### LINKED ISSUE #228: Refactor: Replace custom hooks with rooks
### Type of feature

🧑‍💻 Refactor

### Current behavior

We are using a custom hook for search, but @0-vortex called out the existence of this [package](https://react-hooks.org/docs/useDebounce).

https://github.com/open-sauced/hot/pull/215/files/a6d040095b2eed6863dd527596d00bebb9ceea4b#r928032778

### Suggested solution

1. Replace the [useDebounce](https://github.com/open-sauced/hot/blob/ui-v3-hero/src/hooks/useDebounce.ts) hook with [this](https://react-hooks.org/docs/useDebounce).
2. Leverage this library for this issue https://github.com/open-sauced/hot/issues/205

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by hokagedemehin**
## What type of PR is this? (check all applicable)

- [x] 🧑‍💻 Code Refactor

## Description

This PR addresses the refactor of #228 
I actually only worked on one file (Hero.tsx) but had to resolve some merge conflicts on several other files.
Also, I have some dead codes I am keeping for reference just in case I need to revert back to the way things were so as to understand what I am doing in the first place. These commented codes will be removed once the PR is reviewed and everything works as it should.
## Related Tickets & Documents

Fixes #228 
## Added tests?

- [x] 🙅 no, because they aren't needed

## Added to documentation?

- [x] 🙅 no documentation is needed

## [optional] What gif best describes this PR or how it makes you feel?
iron

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -12,6 +12,9 @@
     <link rel="preconnect" href="https://fonts.googleapis.com">
     <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
     <link href="https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap" rel="stylesheet">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
```
Comment (NsdHSO) [inline index.html:15]: remove this 3 lines

Comment (hokagedemehin) [reply index.html:15]: Screenshot show four lines, which of the lines should I remove.

Is it line 12, 13 and 14

Comment (hokagedemehin) [reply index.html]: I believe I have addressed this request??

**Comment by bdougie** [comment]
Hey @hokagedemehin. I question this statement

> I have some dead codes I am keeping for reference just in case I need to revert back to the way things were so as to understand what I am doing in the first place. These commented codes will be removed once the PR is reviewed and everything works as it should.

You should trust git here and just delete dead code. No need to comment for later. That leads to messy code.

**Comment by bdougie** [comment]
We unfortunately can't test this without PR #215 merged in here. The debounce is leveraged in search and this deploy preview has no search.

**Comment by bdougie** [comment]
Because this is not reviewable, I am going to go ahead and close this. Please do the following. 

1. Start a branch from @shamimbinnur's ui-v3-hero branch
2. Remove all debounce code and replace rooks.
3. Refrain from applying linting changes to other files until code review is completed (do not run `npm run format` before code review)

**Pull request closed without merging**